app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -103,4 +113,50 @@ def format_sync_error(error)
     result << error.backtrace.join("\n")
     result.mb_chars.limit(ERROR_MAX_SIZE)
   end
+
+  private
+
+  # Content borrowed from awx
+  #
+  #  https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L17
+  #
+  VALID_PLAYBOOK_CHECK = /^\s*?-?\s*?(?:hosts|include|import_playbook):\s*?.*?$/
+
+  # Confirms two things:
+  #
+  #   - The file extension is a yaml extension
+  #   - The content of the file has one line that matches VALID_PLAYBOOK_CHECK
+  #
+  # Content borrowed from awx and modified from
+  #
+  #   https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L39-L66
+  #
+  def is_a_playbook?(entry, worktree)
+    return false unless entry[:name] =~ /\.ya?ml$/
+
+    worktree.read_entry(entry).lines.each do |line|
+      return true if line =~ VALID_PLAYBOOK_CHECK
+    end
+
+    return false
+  end
+
+  INVALID_DIRS = %w[roles tasks group_vars host_vars]
+
+  # Given a Pathname, remove the following directories from consideration:
+  #
+  #   - roles/
+  #   - tasks/
+  #   - group_vars/
+  #   - host_vars/
+  #
+  # Also ignores hidden files and directories
+  #
+  # Content borrowed from awx and modified from
+  #
+  #   https://github.com/ansible/awx/blob/128fa894/awx/main/utils/ansible.py#L21-L36
```

Comment:
I feel like the attribution (here and elsewhere) doesn't belong in the code, but instead the commit message.  While we are taking inspiration from awx, we are not trying to be an awx clone, and I don't want to set that expectation.  We may choose to change this code in the future and the attribution sort of forces a contributor to "do it like awx".

Reply:
I could see that.  The only reason I do that is if someone besides me is modifying this in the future, they are more likely to see the comment, and less likely to do a `git blame` to see the commit message.

Reply:
+1 for commit message over comment.

Even the description of the what the method is doing seems overkill ... it's not complicated.

Reply:
Done.  Though, I disagree.